### ISSUE TITLE
macOS support?

### ISSUE THREAD

**Issue by joshsloat** [box/box-ios-sdk#702]
* [X] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/box/box-swift-sdk/blob/master/CONTRIBUTING.md).

## What did you do?

I attempted to integrate a macOS app with the Box SDK.

## What did you expect to happen?

I realize that the name of the SDK is `box-ios-sdk`, but `osx` is listed as one of the supported platforms. As such, I expected the SDK to work with a Mac app. 

![Screen Shot 2020-01-31 at 11 03 40 AM](https://user-images.githubusercontent.com/606159/73554237-5e964900-4419-11ea-928e-74095d6687dc.png)

## What happened instead?

The project links against the SDK without error, but then things start failing silently. I attempted using sample code to authenticate via OAuth and the call never returns. In looking at `obtainAuthorizationCodeFromWebSession`, I discovered that the code simply bails if not iOS. This is a silent fail without any console log if running on another platform. 

```
    func obtainAuthorizationCodeFromWebSession(completion: @escaping Callback<String>) {
        let authorizeURL = makeAuthorizeURL(state: nonce)
        #if os(iOS)
            ... iOS goodness here...
        #endif
    }
```

My preference would be for the SDK to work with Mac (like [Dropbox](https://github.com/dropbox/SwiftyDropbox) does). If Box chooses to not go that direction, the code should be updated to log a console message and/or return an error in the callback and documentation should be updated so others don't spend time going down this same path. 

## BoxSDK Environment

**BoxSDK version:**
3.1.1

**Xcode version:**
11.3.1

**Swift version:**
5

**Platform(s) running BoxSDK:**
macOS

**macOS version running Xcode:**
10.15.2

## Demo Project
[clouduploader.zip](https://github.com/box/box-ios-sdk/files/4140356/clouduploader.zip)
Client secrets and IDs scrubbed from the above for both Box and Dropbox.

**Comment by sujaygarlanka** [comment]
Hello @joshsloat, You are correct in that authentication does not work for platforms other than iOS. If you create a client that takes in a developer token, the SDK should work fine for macOS. We will look into the authentication issue for the other platforms and get back to you soon. Sorry for any confusion.

**Comment by joshsloat** [comment]
Thanks for the response @sujaygarlanka. With the developer token, though, I can only access my account, correct? It's my understanding that I could never build a Mac app for other Box users with that approach (making the SDK not particularly useful for most use cases on Mac).

**Comment by sujaygarlanka** [comment]
@joshsloat, you are correct that you will only be able to access your account. It is really only meant for development purposes. We will update our documentation to reflect that we don't support macOS. If you want support for macOS or any additional features in the future, I would submit feedback [here](https://pulse.box.com/forums/909778-help-shape-the-future-of-box).

**Comment by joshsloat** [comment]
Thanks for the clarification @sujaygarlanka! The link you provided doesn't work for me. Are you suggesting I just submit a new issue for the feature request?

**Comment by sujaygarlanka** [comment]
@joshsloat Sorry, I just updated the link and it should work now. Also, we are looking into adding macOS support in the near future (SDK-450). It is on our backlog.

**Comment by joshsloat** [comment]
Wonderful, @sujaygarlanka. I'll be looking forward to it!

**Comment by Sopan** [comment]
@sujaygarlanka Is it done yet?

**Comment by mgrytsai** [comment]
@Sopan we haven't worked on adding MacOS support yet. It is still in the backlog, but there is no concrete timeline.
